### ISSUE TITLE
[Bug Report] Updates to Gemma

### ISSUE THREAD

**Issue by cmathw** [TransformerLensOrg/TransformerLens#594]
Since support for Gemma was merged [March 14th](https://github.com/TransformerLensOrg/TransformerLens/pull/511), there have been a number of changes to the upstream HF [model file](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gemma/modeling_gemma.py) such that we no longer have good agreement across logits and cache. A large chunk of these changes exist [here](https://github.com/huggingface/transformers/commit/8e2fc52ea3c80d858cc7e7e5f835d4ce795c900d) but I think there are also more recent relevant changes in other commits. I will be opening a corresponding PR to address these issues tomorrow, in the meantime I would not trust Gemma outputs and activations.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)
